### PR TITLE
Probable first attempt at sponsors controller

### PR DESCRIPTION
**Pull request by unkrich**
@kylekthompson @emilyengle #18 
Probably have the auth checks wrong, pls comment

### CODE REVIEW

File: app/controllers/sponsors_controller.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+class SponsorsController < ApplicationController
+  before_action :authorize_user, :except => [:new, :create]
```
Comment (kylekthompson) [inline app/controllers/sponsors_controller.rb]: We're gonna want to authorize for everything (only admins can create and delete, though)

File: app/controllers/sponsors_controller.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  before_action :authorize_admin, :only => [:new, :create, :destroy]
+
```
Comment (kylekthompson) [inline app/controllers/sponsors_controller.rb]: we'll need another `before_action` to verify that the correct user is accessing the information. I have an example of this in `UsersController`

File: app/controllers/sponsors_controller.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  def show
+    @sponsor = Sponsor.find(params[:id])
+  end
+
+  def new
+    @sponsor = Sponsor.new
+  end
+
+  def create
+    @sponsor = Sponsor.new(params[:sponsor])
```
Comment (kylekthompson) [inline app/controllers/sponsors_controller.rb]: We're going to want to use strong params here. Check @emilyengle's PR for an example of how to do this (if you don't know how)

File: app/controllers/sponsors_controller.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    respond_to do |format|
+      if @sponsor.save
+        format.html { redirect_to @sponsor, notice: 'Sponsor was successfully created.' }
+        format.json { render json: @sponsor, status: :created, location: @sponsor }
+      else
+        format.html { render action: "new" }
```
Comment (kylekthompson) [inline app/controllers/sponsors_controller.rb]: this can be `render :new`

File: app/controllers/sponsors_controller.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      end
+    end
+
+  end
+
+  def edit
+    @sponsor = Sponsor.find(params[:id])
+  end
+
+  def update
```
Comment (kylekthompson) [inline app/controllers/sponsors_controller.rb]: this method isn't implemented yet

File: config/routes.rb [hunk 6]
```diff
@@ -1,4 +1,18 @@
 Rails.application.routes.draw do
+  get 'sponsors/index'
```
Comment (kylekthompson) [inline config/routes.rb]: can you use resources for these?

**Comment by unkrich** [comment]
Check update method - might have done that incorrectly.

**Comment by kylekthompson** [comment]
Fixes #18

**Pull request merged**